astroquery/mast/core.py

```diff
@@ -0,0 +1,398 @@
+
+"""
+
+from __future__ import print_function, division
+
+import warnings
+import json
+import time
+import os
+
+import numpy as np
+
+try: # Python 3.x
+    from urllib.parse import quote as urlencode
+except ImportError:  # Python 2.x
+    from urllib import pathname2url as urlencode
+
+from requests import HTTPError
+
+import astropy.units as u
+import astropy.coordinates as coord
+from astropy.table import Table, Row, vstack
+
+from ..query import BaseQuery
+from ..utils import commons, async_to_sync
+from ..utils.class_or_instance import class_or_instance
+from ..exceptions import TimeoutError, InvalidQueryError
+from . import conf
+
+
+__all__ = ['Observations', 'ObservationsClass',
+           'Mast',  'MastClass']
+
+
+class ResolverError(Exception):
+    pass
+
+
+def _prepare_service_request_string(jsonObj):
+    """
+    Takes a mashup json request object and turns it into a url-safe string.
+
+    Parameters
+    ----------
+    jsonObj : dict
+        A Mashup request json object (python dictionary)
+        
+    Returns
+    -------
+    response : str
+        URL encoded Mashup Request string.
+    """
+    requestString = json.dumps(jsonObj)
+    requestString = urlencode(requestString)
+    return "request="+requestString
+
+
+def _mashup_json_to_table(jsonObj):
+    """
+    Takes a json object as returned from a Mashup request and turns it into an astropy Table.
+
+    Parameters
+    ----------
+    jsonObj : dict
+        A Mashup response json object (python dictionary)
+        
+    Returns
+    -------
+    response: `astropy.table.Table`
+    """
+
+    dataTable = Table()
+    
+    if not (jsonObj.get('fields') and jsonObj.get('data')):
+        raise KeyError("Missing required key(s) 'data' and/or 'fields.'")  
+
+    for col,atype in [(x['name'],x['type']) for x in jsonObj['fields']]:
+        if atype=="string":
+            atype="str"
+        if atype=="boolean":
+            atype="bool"
+        dataTable[col] = np.array([x.get(col,None) for x in jsonObj['data']],dtype=atype)
+
+    # Removing "_selected_" column
+    if "_selected_" in dataTable.colnames:
+        dataTable.remove_column("_selected_")
+        
+    return dataTable
+
+
+@async_to_sync
+class MastClass(BaseQuery):
+    """
+    MAST query class.
+
+    Class that allows direct programatic access to the MAST Portal, 
+    more flexible but less user friendly than ObservationsClass.
+    """
+
+    def __init__(self):
+        
+        super().__init__()
+        
+        self._SERVER = conf.server
+        self.TIMEOUT = conf.timeout
+        self.PAGESIZE = conf.pagesize
+
+    
+    def _request(self, method, url, params=None, data=None, headers=None,
+                files=None, stream=False, auth=None, retrieve_all=True, verbose=False):        
+        """
+        Override of the parent method:
+        A generic HTTP request method, similar to `requests.Session.request`
```

Comment:
Well, here it doesn't matter as usually there is no API docs generated from the private methods. Otherwise you need to run the docs build with the ``-w`` option, so any warnings will cause a docs build locally, too.